proc-macro/src/test.rs

```diff
@@ -305,27 +317,45 @@ impl Step {
             event,
             assert,
         } = self;
-        let state = state
+        let (pre, post) = state
             .as_ref()
-            .unwrap_or_else(|| test_state.expect("No state for step"));
-        let State {
-            state_name,
-            state,
-            state_param,
-        } = state;
+            .or(test_state)
+            .map(|state| {
+                let State {
+                state_name,
+                state,
+                state_param,
+                } = state;
+                let state_struct = quote! {
+                    struct State<#(#state_param),*> {
+                        #(#state_name: #state_param,)*
+                    }
+                };
+                let build_struct = quote! {
+                    #(
+                        let #state_name = client.fetch(#state, None).await.unwrap().unwrap();
```

Comment:
Might be better to have `expect` instead of `unwrap` so at least it's easier to debug if the fetch fails.